apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -45,44 +77,7 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     // See CDO implementation at https://github.com/code-dot-org/blockly/blob/main/core/ui/fields/field_dropdown.js#L305
     this.config = element.getAttribute('config');
     if (this.config) {
-      // If `menuGenerator_` is an array, it is an array of options with
-      // each option represented by an array containing 2 elements -
-      // a human-readable string and a language-neutral string. For example,
-      // [['first item', 'ITEM1'], ['second item', 'ITEM2'], ['third item', 'ITEM3']].
-      // Options are included in the block definition.
-      const originalOptionsMap = Array.isArray(this.menuGenerator_)
-        ? this.menuGenerator_.reduce((optionsMap, curr) => {
-            optionsMap[curr[1]] = curr[0];
-            return optionsMap;
-          }, {})
-        : {};
-
-      const numberList = printerStyleNumberRangeToList(this.config);
-      // If numberList is assigned a non-empty array, it contains a list of numbers.
-      // Convert this list to a string of dropdown options separated by commas and assign to options.
-      // Otherwise, assign options to config string.
-      // Note that `config` is either a printer-style number range or a string of options separated
-      // by commas, but not both. For example, a `config` like "1,6-9, &quot;SLOTH&quot;"
-      // would NOT be supported.
-      let options =
-        numberList.length > 0 ? numberListToString(numberList) : this.config;
-
-      options = options.split(',').map(val => {
-        val = val.trim();
-        // If val is one of the options in this.menuGenerator_,
-        // human-readable string is displayed.
-        if (originalOptionsMap[val]) {
-          return [originalOptionsMap[val], val];
-        } else {
-          // Remove quotes and display option with lowercase characters.
-          // For example, "GIRAFFE" would be transformed to giraffe.
-          const humanReadableVal = this.toHumanReadableString(val);
-          return [humanReadableVal, val];
-        }
-      });
-
-      // set the menuGenerator_ to those config options.
-      this.menuGenerator_ = options;
+      this.menuGenerator_ = this.getUpdatedOptionsFromConfig(this.config);
```

Comment:
It feels like an improvement to move this to its own function. Were there any changes in the logic here besides moving it? (Github is making it hard to compare side-by-side.)

Reply:
No logic changes. Initially added since it's now used in both `loadState` and `fromXml` but I think it also makes things a bit easier to read.